apps/swirl-docs/src/components/CodePreview/CodePreviewSelect.tsx

```diff
@@ -0,0 +1,138 @@
+import {
+  SwirlIconCheck,
+  SwirlIconChevronRight,
+} from "@getflip/swirl-components-react";
+import { ComponentProps, useState } from "react";
+import { autoUpdate, flip, shift } from "@floating-ui/dom";
+import {
+  useClick,
+  useDismiss,
+  useFloating,
+  useInteractions,
+  useRole,
+} from "@floating-ui/react";
+import classnames from "classnames";
+import { motion } from "framer-motion";
+
+export type SelectItemProps = ComponentProps<"button"> & {
+  label: string;
+  isSelected: boolean;
+};
+
+export type SelectOption = {
+  label: string;
+  value: string;
+};
+
+export type SelectProps = {
+  options: SelectOption[];
+  selectId: string;
+  onItemClick?: (option: SelectOption) => void;
+};
+
+export function Select({ options, selectId, onItemClick }: SelectProps) {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const { refs, context, x, y } = useFloating({
+    open: isOpen,
+    onOpenChange: setIsOpen,
+    placement: "top-start",
+    middleware: [flip({ fallbackAxisSideDirection: "start" }), shift()],
+    whileElementsMounted: autoUpdate,
+  });
+
+  const click = useClick(context);
+  const dismiss = useDismiss(context);
+  const role = useRole(context);
+
+  const { getReferenceProps, getFloatingProps } = useInteractions([
+    click,
+    dismiss,
+    role,
+  ]);
+
+  const selectedLabel = options.find(
+    (option) => option.value === selectId
+  )?.label;
+
+  return (
+    <div className="relative">
+      <button
+        ref={refs.setReference}
+        className={classnames(
+          "relative flex items-center justify-center",
+          "bg-transparent max-h-5 text-interactive-neutral-default text-font-size-sm",
+          "pr-2",
+          "after:block after:absolute after:h-full after:bg-border-default after:w-[1px] after:opacity-20 after:right-0"
+        )}
+        id="requestlanguage-trigger"
```

Comment:
**issue:** Static id will break if the component is used multiple times on the same page.